File: 1.Iteration - BinaryGap/answer.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(N):
    # write your code in Python 3.6
    
    # Convert to binary
    N = bin(N)[2:]
    binLength = []
    count = 0
    for i in N:
        if (i == str(0)):
            count += 1
        else:
            binLength.append(count)
            count = 0
    return max(binLength)

File: 9.Maximum Slice Problem - MaxProfit/answer.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    max_ending = 0
    max_slice = 0
    let difference = []
    for(let i =0; i < A.length-1; i++){
        let difference = A[i+1] - A[i]
        max_ending = Math.max(0, max_ending+difference)
        max_slice = Math.max(max_slice, max_ending)
    }
    return max_slice
}
